vcd/resource_vcd_vapp_vm.go

```diff
@@ -1943,36 +2087,115 @@ func setGuestProperties(d *schema.ResourceData, properties *types.ProductSection
 // based on the known ID of object.
 //
 // Example resource name (_resource_name_): vcd_vapp_vm.VM_name
-// Example import path (_the_id_string_): org-name.vdc-name.vapp-name.vm-name
+// Example import path for VM within vApp (_the_id_string_): org-name.vdc-name.vapp-name.vm-name
+// Example import path for standalone VM (_the_id_string_): org-name.vdc-name.vm-name
+//
+// The VM identifier can be either the VM name or its ID
+// If we are dealing with standalone VMs, the name can retrieve duplicates. When that happens, the import fails
+// and a list of VM information (ID, guest OS, network, IP) is returned
 func resourceVcdVappVmImport(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
 	resourceURI := strings.Split(d.Id(), ImportSeparator)
-	if len(resourceURI) != 4 {
-		return nil, fmt.Errorf("[VM import] resource name must be specified as org-name.vdc-name.vapp-name.vm-name")
+	vcdClient := meta.(*VCDClient)
+
+	var vapp *govcd.VApp
+	var vm *govcd.VM
+	var orgName string
+	var vdcName string
+	var vappName string
+	var vmIdentifier string
+	standaloneVm := false
+
+	if len(resourceURI) == 3 {
+		// standalone VM
+		orgName, vdcName, vmIdentifier = resourceURI[0], resourceURI[1], resourceURI[2]
+		standaloneVm = true
+	} else {
+		if len(resourceURI) != 4 {
```

Comment:
A comment would be nice similar to above.

Reply:
added